ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -299,18 +299,16 @@ static VALUE stop(VALUE self_instance, VALUE optional_exception) {
 // We need to be careful not to change any state that may be observed OR to restore it if we do. For instance, if anything
 // we do here can set `errno`, then we must be careful to restore the old `errno` after the fact.
 static void handle_sampling_signal(DDTRACE_UNUSED int _signal, DDTRACE_UNUSED siginfo_t *_info, DDTRACE_UNUSED void *_ucontext) {
-  if (!ruby_thread_has_gvl_p()) {
-    return; // Not safe to enqueue a sample from this thread
-  }
-  if (!ddtrace_rb_ractor_main_p()) {
-    return; // We're not on the main Ractor; we currently don't support profiling non-main Ractors
-  }
+  if (
+    !ruby_native_thread_p() || // Not a Ruby thread
+    !is_current_thread_holding_the_gvl() || // Not safe to enqueue a sample from this thread
+    !ddtrace_rb_ractor_main_p() // We're not on the main Ractor; we currently don't support profiling non-main Ractors
```

Comment:
Shoudl that comment be a `// TODO:`?

I was wondering if we should have a categorization system to help focusing on some work, eg: `TODO: [Ractor] blah blah`, so that we can `grep -e 'TODO: [Ractor]' lib` and have all the TODOs on the `Ractor` support topic.

Reply:
Supporting Ractors will need A LOT of changes to the components keeping state to make them safe to be called concurrently. 

Furthermore, Ractors themselves are still quite buggy so I don't think they will be a production thing for several years.

So because of that, the future seems so nebulous that I don't expect any TODOs I leave for now will be useful for later -- we're always going to need to comb through these components and re-evaluate them closely if and when we decide to move to Ractors.